src/auto_dev_loop/add_repo.py

```
@@ -0,0 +1,411 @@
+    if len(projects) == 1:
+        project = projects[0]
+        typer.echo(f"Using project #{project['number']} - {project['title']}")
+        return project
+
+    typer.echo("Available projects:")
+    for i, p in enumerate(projects, 1):
+        typer.echo(f"  {i}. #{p['number']} - {p['title']}")
+    idx = typer.prompt("Select project", type=int)
+    if 1 <= idx <= len(projects):
+        return projects[idx - 1]
+    raise AddRepoError(f"Invalid selection: {idx}")
```

Ruff (0.15.4)

[warning] Avoid specifying long messages outside the exception class

(TRY003)